code/projects/app/src/app/shared/image-comparison/image-comparison.service.ts

```diff
@@ -0,0 +1,49 @@
+import { computed, Injectable, signal } from "@angular/core";
+
+@Injectable({
+  providedIn: "root",
+})
+export class ImageComparisonService {
+  public left = signal<ImageData | undefined>(undefined);
+  public right = signal<ImageData | undefined>(undefined);
+
+  public images = computed(() => {
+    return [this.left(), this.right()];
+  });
+
+  public addImage(which: "left" | "right", image: ImageData): Promise<boolean> {
+    // Resolve false only when the image is not the same size as the other image
+    const promise = new Promise<boolean>((resolve) => {
```

Comment:
I can't understand why this code returns a promise.
I don't see any async operations or callbacks inside, is this left over from some previous experimentation? I can remove the `new Promise` wrapper, and make the function return `: boolean` without any problems locally.

If the callers of this method needs to take a promise, let them wrap it instead, but it seems like no callers actually require a promise anywhere.

Reply:
You are right, propably was left while experimenting, I can't recall why I did it. Thansk for pointing it out.